plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/GcpLocalRunTab.java

```diff
@@ -206,14 +229,28 @@ public void modifyText(ModifyEvent event) {
     String[] filterExtensions = new String[] {"*.json"}; //$NON-NLS-1$
     browse.addSelectionListener(new FileFieldSetter(serviceKeyInput, filterExtensions));
 
+    serviceKeyDecoration = new ControlDecoration(serviceKeyInput, SWT.LEAD | SWT.TOP);
+    serviceKeyDecoration.hide();
+
+    createServiceKey = new Button(composite, SWT.NONE);
```

Comment:
Would it be worth having a hover on this button to explain that it creates a new App Engine service account key, same as what the app will be executed on App Engine?  Just in case they have multiple service accounts…

Reply:
Good idea. Personally, I think all of the controls in this UI needs more explanation.